integration-test/src/test/java/org/commonjava/maven/ext/manip/CliTestUtils.java

```diff
@@ -119,99 +124,46 @@ public static void runLikeInvoker( String workingDir, String restURL )
      *
      * @param args - List of additional command line arguments
-     * @param workingDir - Working directory in which you want the cli to be run.
-     * @return Exit value
-     * @throws Exception
-     */
-    public static Integer runCli( List<String> args, String workingDir )
-        throws Exception
-    {
-        Properties testProperties = Utils.loadProps( workingDir + "/test.properties" );
-        Map<String, String> javaParams = Utils.propsToMap( testProperties );
-        return runCli( args, javaParams, workingDir );
-    }
-
-    /**
-     * Run pom-manipulation-cli.jar with java params (-D arguments) in workingDir directory.
-     *
      * @param params - Map of String keys and String values representing -D arguments
      * @param workingDir - Working directory in which you want the cli to be run.
      * @return Exit value
      * @throws Exception
      */
-    public static Integer runCli( Map<String, String> params, String workingDir )
+    private static Integer runCli( List<String> args, Map<String, String> params, String workingDir )
         throws Exception
     {
-        return runCli( null, params, workingDir );
-    }
+        ArrayList<String> arguments = new ArrayList<String>( args );
+        Collections.addAll( arguments, toJavaParams( params ).split( "\\s+" ) );
+        arguments.add( "--log=" + workingDir + File.separator + "build.log" );
+        arguments.add( "--file=" + workingDir + File.separator + "pom.xml" );
 
-    /**
-     * Run pom-manipulation-cli.jar with java params (-D arguments) in workingDir directory.
-     *
-     * @param args - List of additional command line arguments
-     * @param params - Map of String keys and String values representing -D arguments
-     * @param workingDir - Working directory in which you want the cli to be run.
-     * @return Exit value
-     * @throws Exception
-     */
-    public static Integer runCli( List<String> args, Map<String, String> params, String workingDir )
-        throws Exception
-    {
-        String stringArgs = toArguments( args );
-        String stringParams = toJavaParams( params );
-        String command =
-            String.format( "%s -jar %s/pom-manipulation-cli.jar %s %s", getJavaBin().getAbsolutePath(), BUILD_DIR, stringParams, stringArgs );
+        logger.info( "Invoking CLI with {} ", arguments );
+        int result = new Cli().run( arguments.toArray( new String[arguments.size()] ) );
 
-        return runCommandAndWait( command, workingDir, null );
-    }
+        // This is a bit of a hack. The CLI, if log-to-file is enabled resets the logging. As we don't fork and run
+        // in the same process this means we need to reset it back again. The benefit of not forking is a simpler test
+        // harness and it saves time when running the tests.
+        final ch.qos.logback.classic.Logger root = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger( org.slf4j.Logger.ROOT_LOGGER_NAME );
 
```

Comment:
Avoid forking the CLI and just call inline. However, it does mean I need to reset the logging after....